File: RetrofitJava/app/src/main/java/com/tolgaalperkus/retrofitjava/view/MainActivity.java
package com.tolgaalperkus.retrofitjava.view;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.tolgaalperkus.retrofitjava.R;
import com.tolgaalperkus.retrofitjava.adapter.RecyclerViewAdapter;
import com.tolgaalperkus.retrofitjava.model.CryptoModel;
import com.tolgaalperkus.retrofitjava.service.CryptoAPI;

import java.util.ArrayList;
import java.util.List;

import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.schedulers.Schedulers;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

public class MainActivity extends AppCompatActivity {

    ArrayList<CryptoModel> cryptoModels;
    private String BASE_URL = "https://api.nomics.com/v1/";
    Retrofit retrofit;
    RecyclerView recyclerView;
    RecyclerViewAdapter recyclerViewAdapter;
    CompositeDisposable compositeDisposable;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        recyclerView = findViewById(R.id.recyclerView);

        //https://api.nomics.com/v1/prices?key=297a3e6af3c71f34b3b9668232a5ff37

        Gson gson = new GsonBuilder().setLenient().create();

        retrofit = new Retrofit.Builder()
                .baseUrl(BASE_URL)
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .addConverterFactory(GsonConverterFactory.create(gson))
                .build();
        loadData();
    }

    private void loadData(){
        CryptoAPI cryptoAPI = retrofit.create(CryptoAPI.class);
        compositeDisposable = new CompositeDisposable();
        compositeDisposable.add(cryptoAPI.getData().subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe(this::handleResponse));
        /*
        Call<List<CryptoModel>> call = cryptoAPI.getData();
        call.enqueue(new Callback<List<CryptoModel>>() {
            @Override
            public void onResponse(Call<List<CryptoModel>> call, Response<List<CryptoModel>> response) {
                if(response.isSuccessful()){
                    List<CryptoModel> responseList = response.body();
                    cryptoModels = new ArrayList<>(responseList);
                    //RecyclerView
                    recyclerView.setLayoutManager(new LinearLayoutManager(MainActivity.this));
                    recyclerViewAdapter = new RecyclerViewAdapter(cryptoModels);
                    recyclerView.setAdapter(recyclerViewAdapter);

                }

            }

            @Override
            public void onFailure(Call<List<CryptoModel>> call, Throwable t) {
                t.printStackTrace();
            }
        });*/
    }
    private void handleResponse(List<CryptoModel> cryptoModelList){
        cryptoModels = new ArrayList<>(cryptoModelList);
        recyclerView.setLayoutManager(new LinearLayoutManager(MainActivity.this));
        recyclerViewAdapter = new RecyclerViewAdapter(cryptoModels);
        recyclerView.setAdapter(recyclerViewAdapter);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        compositeDisposable.clear();
    }
}

File: KotlinSimpleCalculator/app/src/main/java/com/tolgaalperkus/kotlinsimplecalculator/MainActivity.kt
package com.tolgaalperkus.kotlinsimplecalculator

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var firstNumber : Int? = null
    var secondNumber : Int? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun mySum (view : View) {
        firstNumber = editText.text.toString().toIntOrNull()
        secondNumber = editText2.text.toString().toIntOrNull()
        if(firstNumber != null && secondNumber != null){
            resultText.setText("Result :" + (firstNumber!!+secondNumber!!))
        }
        else{
            resultText.setText("Do not leave empty")
        }
    }

    fun mySub (view:View){
        firstNumber = editText.text.toString().toIntOrNull()
        secondNumber = editText2.text.toString().toIntOrNull()
        if(firstNumber != null && secondNumber != null){
            resultText.setText("Result :" + (firstNumber!!-secondNumber!!))
        }
        else{
            resultText.setText("Do not leave empty")
        }

    }

    fun myMultiply(view:View){
        firstNumber = editText.text.toString().toIntOrNull()
        secondNumber = editText2.text.toString().toIntOrNull()
        if(firstNumber != null && secondNumber != null){
            resultText.setText("Result :" + (firstNumber!!*secondNumber!!))
        }
        else{
            resultText.setText("Do not leave empty")
        }

    }

    fun myDiv(view:View){
        firstNumber = editText.text.toString().toIntOrNull()
        secondNumber = editText2.text.toString().toIntOrNull()
        if(firstNumber != null && secondNumber != null){
            resultText.setText("Result :" + (firstNumber!!/secondNumber!!))
        }
        else{
            resultText.setText("Do not leave empty")
        }

    }
}